overlord/configstate/config/transaction.go

```diff
@@ -290,7 +307,11 @@ func (t *Transaction) Commit() {
 		if !ok {
 			config = make(map[string]*json.RawMessage)
 		}
-		applyChanges(config, snapChanges)
+		applyChanges(instanceName, config, snapChanges)
+
+		// XXX: do not pass instance name
+		t.purgeHijacked(instanceName, config, nil)
```

Comment:
this is a bit strange I would expect to do this before applying or together, but also for this to work we need to set up the hijacking also during set commands but then the code in corecfg will get confused because it will try the hijack code while trying to find out what has changed afaict which we don't want.  It seems like we need more variations on Get ? This needs more testing and thinking, also about what to do about snapshots and configuration and also reverts.